lib/Gocdb_Services/ServiceService.php

```diff
@@ -1876,6 +1894,50 @@ public function deleteEndpoint(\EndpointLocation $endpoint, \User $user) {
         }
     }
 
+    /**
+     * Function to establish if an endpoint already has a value sets
+     *
+     * @returns boolean
+    */
+    public function EndpointPropSet (\EndpointLocation $endpoint, $endpointPropName) {
+      switch (strtolower($endpointPropName)) {
+        case 'name':{
+          $propValue = $endpoint->getName();
+          break;
+        }
+        case 'url':{
+          $propValue = $endpoint->getUrl();
+          break;
+        }
+        case 'interfacename':{
+          $propValue = $endpoint->getInterfaceName();
+          break;
+        }
+        case 'description':{
+          $propValue = $endpoint->getDescription();
+          break;
+        }
+        case 'email':{
+          $propValue = $endpoint->getEmail();
+          break;
+        }
+        case 'monitored': {
+          #booleans are always set
+          return true;
+        }
+        default:{
+          throw new \Exception("Internal error: endpoint property name ($endpointPropName) not ".
+          "recognised. Please contact a GOCDB administrator and report this error.");
+        }
+      }
+
+      if (empty($propValue)) {
+        return false;
+      } else {
+        return true;
+      }
```

Comment:
reverse logic for readability

Reply:
for clarity, what I meant was replacing the `if`/`else` statement with something like `return ! empty($propValue)`